tests/test_image_upload.py

```diff
@@ -61,6 +64,18 @@
 ]
 
 
+@pytest.fixture(scope="module")
```

Comment:
all of the bellow fixtures should go to conftest.py, so they can be reused by other tests, if there is an assumption that they are too general and might get reused. fixtures switching and creating test_dirs are definitely too. Also, you should go through pytest documentation first, coz pytest already offers everything you are trying to reinvent here, it offers multiple fixtures for creating or switching test_dirs/files.

Reply:
We already have a fixture for this, I just needed it to be module-scoped. Ended up not working the way I wanted, so now I just use said fixture directly.

Reply:
Why did you need it to be module-scoped?

Reply:
Because the fixture using that one was module-scoped, it was module-scoped because function scope was getting applied once each parametrized run, or maybe it was already parallelization problems and I just didn't know that yet and was hoping module scope would solve it? Honestly don't know at this point. Doesn't matter, it had no chance of working in hindsight.

Reply:
I understand the parallelization pain.  It can be solved by using tmp_path, but it this case it probably makes more sense to pre-create the files as you did.